src/workflow.ts

```diff
@@ -1,84 +1,216 @@
 import { Graph } from "graph-data-structure"
+import { z } from "zod"
 
-import { typedEmitter } from "./events"
+import { typedEmitter } from "./events.js"
 
-/**
- * This is a helper type to create a strongly typed workflow definition.
- *
- * You define a workflow as a set of tasks with return types, and provide
- * a type for a context object that is passed to each task when it is run.
- *
- * I was hoping to infer or decentralize the return type definitions, but
- * that just lead to circular type definitions as TypeScript is not able to
- * partially fulfill a generic type to resolve the circular dependency.
- */
-export type WorkflowDefinition<
-  T extends { context: unknown; tasks: Record<string, unknown> } = {
-    context: unknown
-    tasks: Record<string, unknown>
-  },
-> = T
+interface UnknownWorkflowDefinition {
+  context: unknown
```

Comment:
trying to find a way to get this context back to the build workflow function without losing type inference

Reply:
I can make it work but only by removing the zod validators: https://github.com/sargunv/async-task-graph/pull/4